docs/developer/stock/overview.mdx

```diff
@@ -0,0 +1,321 @@
+
+```graphql {2-9}
+mutation {
+  createWarehouse(
+    input: {
+      name: "Example warehouse", 
+      slug: "example-warehouse", 
+      email: "example-warehouse@example.com", 
+      externalReference: "Warehouse_001", 
+      address: {streetAddress1: "Teczowa 8", city: "Wroclaw", country: PL, postalCode: "53-601", companyName: "Amazing Company Inc"}
+    }
+  ) {
+    warehouse {
+      id
+      name
+      slug
+      email
+      externalReference
+      address {
+        id
+        companyName
+      }
+    }
+    errors {
+      message
+      field
+      code
+    }
+  }
+}
+```
+
+As a response we get the newly created warehouse:
+
+```json
+{
+  "data": {
+    "createWarehouse": {
+      "warehouse": {
+        "id": "V2FyZWhvdXNlOmY1MTJiMzFjLTUyMzAtNDMwOS05MDdkLTY4NzAyY2NhMjg0MA==",
+        "name": "Example warehouse",
+        "slug": "example-warehouse",
+        "email": "example-warehouse@example.com",
+        "externalReference": "Warehouse_001",
+        "address": {
+          "id": "QWRkcmVzczozNjcx",
+          "companyName": "Amazing Company Inc"
+        }
+      },
+      "errors": []
+    }
+  }
+}
+```
+
+### Warehouse Update
+Once created, following warehouse fields can be updated:
+- name
+- slug
+- email
+- address
+- externalReference
+- isPrivate
+- clickAndCollectOption
+
+:::info
+Only public warehouses can later be configured as a collection point.
+:::
+You can update warehouse using Saleor Dashboard or GraphQL API [`updateWarehouse`](api-reference/products/mutations/update-warehouse.mdx) mutation.
+
+To manage warehouse metadata, use [`updateMetadata`](/api-reference/miscellaneous/mutations/update-metadata.mdx) 
+or [`updatePrivateMetadata`](/api-reference/miscellaneous/mutations/update-private-metadata.mdx). 
+
+### Assign Warehouses to the Channels and Shipping
+
+To make a warehouse available for sales and shipping, it must be assigned 
+to the appropriate **sales channels** and **shipping zones**.
+
+#### Assign Warehouses to the Channel
+The first step is to assign your warehouses to the relevant sales channels using Dashboard or GraphQL API
+
+**Saleor Dashboard**
+1. Go to Configuration → Channels.
+2. Select the channel you want to configure.
+3. On the right panel, find the Warehouses section.
+4. Click Add Warehouse.
+5. Select a warehouse from the dropdown or search by name.
+6. Click Save Channel.
+
+**API**
+
+Use the [`channelUpdate`](api-reference/channels/mutations/channel-update.mdx) mutation to add a warehouse to a channel:
+
+```graphql {2-6}
+  mutation {
+    channelUpdate(
+      id: "Q2hhbm5lbDoxMQ==",
+      input: {
+        "addWarehouses": ["V2FyZWhvdXNlOjA4N2I2MzhlLWIxYzYtNDk5Mi1hMWQ1LTZiOWI5N2FjODU1Zg=="]
+      }
+    ){
+      channel{
+        id
+        warehouses {
+          slug
+        }
+      }
+      errors{
+        field
+        code
+        message
+        shippingZones
+        warehouses
+      }
```

Comment:
Please paste the queries into the GraphQL Playground and prettify them, the brackets are not aligned 🧐.